src/clientlist.cpp

```diff
@@ -399,6 +399,16 @@ void unmanage_client(Window win) {
     if (m) monitor_apply_layout(m);
     ewmh_remove_client(win);
     tag_set_flags_dirty();
+
+    // Get the current client and update the windows focus.
+    client = frame_focused_client(tag->frame);
+    if (client) {
+        client_window_focus(client);
+    } else {
+        char buf[STRING_BUF_SIZE];
+        snprintf(buf, STRING_BUF_SIZE, "0x%x", (unsigned int) 0);
+        hook_emit_list("window_title_changed", buf, "", NULL);
```

Comment:
Thanks for your work! IMO, the final diff looks very good. :-)

Just some non-technical/non-programming comments:

Could you please merge the two commits into one? I'd like that because
its not yet in master, and I think we should avoid adding code and
removing it directly afterwards :-) (i.e. in those cases, we can just
omit that code from our history) :-)

Oh and as The-Compiler just spotted, add curly braces around the single
statement in your then-clause:

  if (!client) {
     hook_emit_list("window_title_changed", NULL);
  }